### PR TITLE
[APR] Add additional arg to SearchPoolForm to filter GQL

### PR DESCRIPTION
**Pull request by Rawallon**
Preview link: https://tools-monorepo-2i09av0rl-bleu-fi.vercel.app/
Preview link for pool simulator: https://tools-monorepo-2i09av0rl-bleu-fi.vercel.app/poolsimulator

This PR:
- adds gqlAdditionalVariable to SearchPoolForm
- Refactor Pool simulator SearchPoolForm to get pools by type using gqlAdditionalVariable
- on APR gets only pools that have the TVL greater then 5k

### CODE REVIEW

File: apps/balancer-tools/src/app/apr/(components)/HeaderEndButton.tsx [hunk 1]
```diff
@@ -59,7 +59,7 @@ export default function HeaderEndButton() {
             defaultValueNetwork={networkIdFor(network as string)}
             onSubmit={handlePoolClick}
             showPools
-            onlyVotingGauges
+            gqlAdditionalVariable={{ totalLiquidity_gt: 5000 }}
```
Comment (luizakp) [inline apps/balancer-tools/src/app/apr/(components)/HeaderEndButton.tsx:62]: Awesome!

File: apps/balancer-tools/src/components/SearchPoolForm/index.tsx [hunk 2]
```diff
@@ -73,7 +73,8 @@ export function SearchPoolForm({
   const { data: poolsDataList, mutate: poolsDataListMutate } = pools
     .gql(network || "1")
     .usePoolsWherePoolType(
-      poolTypeFilter?.length ? { poolTypes: poolTypeFilter } : {},
+      gqlAdditionalVariable,
+      // poolTypeFilter?.length ? { poolTypes: poolTypeFilter } : {},
```
Comment (luizakp) [inline apps/balancer-tools/src/components/SearchPoolForm/index.tsx:77]:
```suggestion

```

File: apps/balancer-tools/src/components/SearchPoolForm/index.tsx [hunk 3]
```diff
@@ -191,8 +192,10 @@ export function SearchPoolForm({
                         <span>
                           {/* By default, FX pools have the name equal to "BPT".
                             So, we'll use name instead of the symbol, since it is more meaningful */}
-                          {poolTypeFilter?.[0] == "FX" &&
-                          poolTypeFilter?.length == 1
+                          {gqlAdditionalVariable?.poolTypeFilter &&
+                          Array.isArray(gqlAdditionalVariable.poolTypeFilter) &&
+                          gqlAdditionalVariable.poolTypeFilter?.[0] == "FX" &&
+                          gqlAdditionalVariable.poolTypeFilter?.length == 1
```
Comment (luizakp) [inline apps/balancer-tools/src/components/SearchPoolForm/index.tsx:198]: Something here broke the name edge case for FX

Production:
image

This branch preview

image

**Review by luizakp — COMMENTED** [review]
I'll fix the name thing, then ready to merge

File: apps/balancer-tools/src/app/apr/(components)/HeaderEndButton.tsx (hunk 1 above)
Comment (ribeirojose) [reply apps/balancer-tools/src/app/apr/(components)/HeaderEndButton.tsx:62]: 👌

**Review by yvesfracari — COMMENTED** [review]
Great refactor! Looking sharp

**Pull request merged**